e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -45,6 +49,10 @@ public void When_extractRequestReceived_Expect_ExtractStatusAddedToDatabase() th
         assertThat(ehrRequest.get("toAsid")).isEqualTo(TO_ASID);
         assertThat(ehrRequest.get("fromOdsCode")).isEqualTo(FROM_ODS_CODE);
         assertThat(ehrRequest.get("toOdsCode")).isEqualTo(TO_ODS_CODE);
+        var accessStructured = (Document) ehrExtractStatus.get(GPC_ACCESS_STRUCTURED);
```

Comment:
What if we split our test/assertion into two helper methods:
assertThatInitialRecordWasCreated
assertThatAccessStructuredWasFetched

within `assertThatAccessStructuredWasFetched` we can have a second `waitFor(() -> ` that waits until the ehrRequest exists.